man/fi_tcp.7.md

```diff
@@ -90,6 +90,24 @@ show all environment variables defined for the tcp provider.
   through the standard socket APIs (i.e. connect, accept, send, recv).
   Default: disabled.
 
+The tcp provider also supports setting parameters via fi_set_val.
+
+*FI_TCP_FABRIC_PASSIVE_PORT_RANGE*
+: Set values corresponding to FI_TCP_PORT_LOW_RANGE and FI_TCP_PORT_HIGH_RANGE.
+  Use fabric object with struct xnet_port_range * set with desired values.
+  Sample:
+    struct xnet_port_range range = { .low = 9000, .high = 9100 };
+    fi_set_val(&fabric->fid, FI_TCP_FABRIC_PASSIVE_PORT_RANGE, &range);
```

Comment:
The structure should be defined in a public header file available to the user.  The standard practice for this would be to define prov/tcp/include/rdma/fi_ext_tcp.h, and define a structure there.  The structure name should probably be something like fi_tcp_port_range { }.  The new header should be installed if the tcp provider is built.

As an example, you can look at the gni provider.  See prov/gni/include/fi_ext_gni.h.  (I think they should have placed the file under include/rdma.)

Reply:
Btw, if the structure is defined as:
```
struct fi_tcp_port_range {
    uint16_t low;
    uint16_t high;
};
```
then you can remove the checks for the port being out of range.